graphql_compiler/macros/macro_edge/validation.py

```diff
@@ -99,6 +102,37 @@ def _validate_macro_edge_name_for_class_name(schema, class_name, macro_edge_name
             .format(macro_edge_name, class_name))
 
 
+def _get_minimal_query_ast_from_macro_ast(macro_ast):
+    """Get a query that should successfully compile to IR if the macro is valid."""
+    macro_directives = get_directives_for_ast(macro_ast)
+    query_ast = remove_directives_from_ast(macro_ast, {
+        directive.name
+        for directive in MACRO_EDGE_DIRECTIVES
+    })
+
+    # Add an output directive to make the ast a valid query
+    if 'output' not in macro_directives:
+        output_directive = Directive(Name('output'), arguments=[
+            Argument(Name('out_name'), StringValue('dummy_output_name'))
+        ])
+
+        # Add an output to a uuid field at the beginning of the query
+        # HACK(bojanserafimov): It's possible that the type at the top doesn't have a uuid field.
+        #                       It's also possible that none of the types in the macro have a field
+        #                       that we can output. The proper solution is to bypass the rule that
+        #                       a query has an output.
+        field = None
+        top_level_selections = query_ast.selection_set.selections[0].selection_set.selections
+        for selection in top_level_selections:
+            if isinstance(selection, Field):
+                if selection.name.value == 'uuid':
+                    field = selection
+        if field is None:
+            top_level_selections.insert(0, Field(Name('uuid'), directives=[]))
```

Comment:
We seem to be setting `directives=[]` and then below mutating it with `field.directives.append` -- why not construct it fully the first time around?

Reply:
It makes the logic simpler in the case when the field already exists. Will restructure to make it obvious.